lib/segment/src/index/field_index/bool_index/mmap_bool_index.rs

```diff
@@ -223,59 +252,86 @@ impl MmapBoolIndex {
         is_true: bool,
         hw_counter: &HardwareCounterCell,
     ) -> bool {
+        let Some(storage) = &self.storage else {
+            return false;
+        };
+
         let hw_counter = self.make_conditioned_counter(hw_counter);
 
         hw_counter
             .payload_index_io_read_counter()
             .incr_delta(size_of::<bool>());
         if is_true {
-            self.trues_slice.get(point_id as usize)
+            storage.trues_slice.get(point_id as usize)
         } else {
-            self.falses_slice.get(point_id as usize)
+            storage.falses_slice.get(point_id as usize)
         }
     }
 
     pub fn values_is_empty(&self, point_id: u32) -> bool {
-        !self.trues_slice.get(point_id as usize) && !self.falses_slice.get(point_id as usize)
+        let Some(storage) = &self.storage else {
+            return true;
+        };
+
+        !storage.trues_slice.get(point_id as usize) && !storage.falses_slice.get(point_id as usize)
     }
 
     pub fn iter_values_map<'a>(
         &'a self,
         hw_counter: &'a HardwareCounterCell,
-    ) -> impl Iterator<Item = (bool, IdIter<'a>)> + 'a {
+    ) -> Box<dyn Iterator<Item = (bool, IdIter<'a>)> + 'a> {
         let hw_counter = self.make_conditioned_counter(hw_counter);
 
-        [
-            (false, Box::new(self.falses_slice.iter_trues()) as IdIter),
-            (true, Box::new(self.trues_slice.iter_trues()) as IdIter),
+        let Some(storage) = &self.storage else {
+            return Box::new(std::iter::empty());
+        };
+
+        let iter = [
+            (false, Box::new(storage.falses_slice.iter_trues()) as IdIter),
+            (true, Box::new(storage.trues_slice.iter_trues()) as IdIter),
         ]
         .into_iter()
         .measure_hw_with_acc(hw_counter.new_accumulator(), u8::BITS as usize, |i| {
             i.payload_index_io_read_counter()
-        })
+        });
+        Box::new(iter)
     }
 
-    pub fn iter_values(&self) -> impl Iterator<Item = bool> + '_ {
-        [
-            self.falses_slice.iter_trues().next().map(|_| false),
-            self.trues_slice.iter_trues().next().map(|_| true),
+    pub fn iter_values(&self) -> Box<dyn Iterator<Item = bool> + '_> {
```

Comment:
Let's use Either to keep impl instead of Box
or
Add a comment if this is temporary